File: src/main/java/com/baulsupp/oksocial/completion/ApiCompleter.java
package com.baulsupp.oksocial.completion;

import java.io.IOException;
import java.util.concurrent.CompletableFuture;
import okhttp3.HttpUrl;

public interface ApiCompleter {
  CompletableFuture<UrlList> prefixUrls() throws IOException;

  CompletableFuture<UrlList> siteUrls(HttpUrl url) throws IOException;
}

File: src/main/kotlin/com/baulsupp/oksocial/completion/ApiCompleter.kt
package com.baulsupp.oksocial.completion

import com.baulsupp.oksocial.credentials.Token
import okhttp3.HttpUrl

interface ApiCompleter {
  /**
   * Top level, usually host level completion.
   */
  suspend fun prefixUrls(): UrlList

  /** Site specific url completion usually within a single service e.g. https://api.twitter.com/ */
  suspend fun siteUrls(url: HttpUrl, tokenSet: Token): UrlList
}
